### PR TITLE
Update date_time.html.twig better check for emptiness

### LINKED ISSUE #33090: Unexpected behaviour with DateTimeColumn empty value
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When you use a DateTimeColumn in a Grid, you're able to define a value to display in case the DateTimeColumn value would be empty, thanks to the empty_data parameter. Empty_data parameter is by default set to an empty string.

When passing an empty value to this column type, the current datetime is displayed.

### Expected behavior

The expected behaviour, is that if the DateTimeColumn receives an empty data, it displays an empty string, or the defined empty_data parameter's value.

### Steps to reproduce

Add or use any DateTimeColumn in a Grid. 
Pass it an empty value (false, null, 0, "", ...).
Check that it renders the current datetime instead of the empty_data parameter's value.

### PrestaShop version(s) where the bug happened

latest

### PHP version(s) where the bug happened

any

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

Evolutive Group

### PR DESCRIPTION
**Pull request by Kaikina**
| Questions              | Answers
| -----------------      | -------------------------------------------------------
| Branch?                | develop 
| Description?           | Date_time column type allows to declare a specific data to display in case its data would be empty. However, it only checks if the date is a default value ('0000-00-00 00:00:00') and not if it is "empty" (false, null, "", ...). This PR adds a emptiness check in the template, as giving an empty data would result in the current datetime being displayed and this is not correct.
| Type?                  | bug fix 
| Category?              | BO
| BC breaks?             | no
| Deprecations?          | no
| How to test?           | In the OrderQueryBuilder class, in the getSearchQueryBuilder, replace 'o.date_add' by 'null as date_add'. Now you can check that it displays an empty string instead of current datetime.
| Fixed ticket?          | Fixes https://github.com/PrestaShop/PrestaShop/issues/33090
| Related PRs            | NA
| Sponsor company        | Evolutive Group

### CODE REVIEW

**Review by eternoendless — APPROVED** [review]
LGTM

**Review by jolelievre — APPROVED** [review]

**Comment by jolelievre** [comment]
Thanks @Kaikina

**Comment by jolelievre** [comment]
I think this QA should be done by a dev, and automatic tests (including UI tests) should be enough to ensure this doesn't introduce a regression

**Comment by jolelievre** [comment]
Running campaign https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/5452826454

**Review by PululuK — APPROVED** [review]
thanks @Kaikina :)

**Review by boherm — APPROVED** [review]

**Comment by nicosomb** [comment]
As we have 2 devs (@boherm and @PululuK) who approve the PR after the "waiting for dev", is that OK for you to merge?

**Comment by boherm** [comment]
> As we have 2 devs (@boherm and @PululuK) who approve the PR after the "waiting for dev", is that OK for you to merge?

I didn't test this PR on my side. I could test only tomorrow.

**Review by M0rgan01 — APPROVED** [review]

**Comment by kpodemski** [comment]
tested by @M0rgan01 

thanks @Kaikina for your PR

**Pull request merged**
